hathor/consensus.py

```diff
@@ -119,14 +121,22 @@ def update(self, base: BaseTransaction) -> None:
                               count=len(to_remove))
                 storage.remove_transactions(to_remove)
                 for tx_removed in to_remove:
-                    self.pubsub.publish(HathorEvents.CONSENSUS_TX_REMOVED, tx_hash=tx_removed.hash)
+                    context.pubsub.publish(HathorEvents.CONSENSUS_TX_REMOVED, tx_hash=tx_removed.hash)
 
         # finally signal an index update for all affected transactions
         for tx_affected in context.txs_affected:
             assert tx_affected.storage is not None
             assert tx_affected.storage.indexes is not None
             tx_affected.storage.indexes.update(tx_affected)
-            self.pubsub.publish(HathorEvents.CONSENSUS_TX_UPDATE, tx=tx_affected)
+            context.pubsub.publish(HathorEvents.CONSENSUS_TX_UPDATE, tx=tx_affected)
+
+        # and also emit the reorg event if needed
+        if context.reorg_started:
+            old_best_block = base.storage.get_transaction(best_tip)
+            new_best_block = base.storage.get_transaction(new_best_tip)
+            context.pubsub.publish(HathorEvents.REORG_FINISHED, old_best_height=best_height,
```

Comment:
Should we assert that `old_best_block != new_best_block`?

Reply:
I think that could make sense, but is that a guarantee that we already have?

Reply:
I tried that and several test cases fail, so it doesn't look like we have that guarantee. I think it should be fine to live with this because the alternatives would be buffering events and only emit REORG_STARTED and start a group-id if it turns out there actually was a re-org, _or_ figure out how to change the implementation so that we actually have that guarantee.

Reply:
So we might have zero-length reorgs? Is it the case when we have to handle ties? I'm still figuring out when it happens.

Reply:
I'm not sure if it's only related to ties.

Also, I'm considering removing the `reorg_size` field, I don't think it has (1) a clear definition (I chose the height difference, but if there's a reorg from genesis that reaches the same height `reorg_size` will be 0, which might be really misleading), and maybe more importantly (2) an application (if there is any it's not obvious and it wasn't mentioned on the design).

Reply:
As discussed, `reorg_size` has a clear definition: It is the number of blocks that belonged to the best blockchain and became voided. The only benefit of adding this to the event is that `reorg_size` is a well-known parameters used by exchanges and use cases.

Reply:
Just fixed all issues related to detecting a reorg event. So, I'm marking this conversation as resolved.